03-Bash/kernel_git/kernel_stats.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+if [ $# -ne 1 ]; then
+    echo "Incorrect usage."
+    echo "Path to the local Linux source repository is required."
+    exit
+elif [ ! -d $1 ]; then
+    echo "$1 is not a directory."
+    exit
+fi
+
+cd $1
+
+if [ ! -d .git ]; then
+    echo "$1 is not a git repository directory."
+    exit
+fi
+
+echo "Repository contains:"
+echo "$(git ls-files '*.c' | wc -l) *.c files;"
+echo "$(git ls-files '*.cpp' | wc -l) *.cpp files;"
+echo "$(git ls-files '*.py' | wc -l) *.py files."
+echo ""
+
+REV_NUM=$(git log --oneline | grep -c revert)
+echo "There are $REV_NUM commits with word 'revert' in the name of the commit."
+echo ""
+
+echo "Number of lines of code in *.py files by author:"
+git ls-files '*.py' | \
+    while read f; do
+        echo -e "File $f:"; git blame --line-porcelain $f | grep '^author ' | \
```

Comment:
This script get info about lines in each file but the quantity for all files together is requested.

Reply:
Done.